generators/newfpmpage/index.js

```diff
@@ -0,0 +1,117 @@
+                type: "input",
+                name: "viewName",
+                message: "What is the name of the page view?",
+                validate: (s) => {
+                    if (/^\d*[a-zA-Z][a-zA-Z0-9]*$/g.test(s)) {
+                        return true;
+                    }
+                    return "Please use alpha numeric characters only for the view name.";
+                },
+                default: 'Main'
+            },
+            
+        ]);
+
+        const manifest = this.fs.readJSON(this.destinationPath(this.answers.moduleName || this.options.modulename, 'webapp', 'manifest.json'));
+        if (!manifest) {
+            this.answers.serviceUrl = (await this.prompt([{
+                type: 'input',
+                name: 'serviceUrl',
+                message: 'What is the url of the main service?',
+                default: 'https://iccsrm.sap.com:44300/sap/opu/odata4/iwbep/v4_sample/default/iwbep/v4_gw_sample_basic/0001',
+                validate: (s) => !!s
+            }])).serviceUrl;
+        
+            const url = new URL(this.answers.serviceUrl);
+            this.answers.host = url.origin;
+            this.answers.path = url.pathname;
+            const service = axios.createServiceForUrl(this.answers.serviceUrl, {
+                ignoreCertErrors: true
+            });
+    
+            while (!this.answers.metadata) {
+                try {
+                    this.answers.metadata = await service.metadata();
+                } catch (error) {
+                    if (service.defaults?.auth?.username) {
+                        this.log.error(error.cause.statusText);
+                    }
+                    if (error.cause.status === 401) {
+                        const { username, password } = await this.prompt([
+                            {
+                                type: 'input',
+                                name: 'username',
+                                message: 'Username',
+                                validate: (answer) => !!answer
+                            },
+                            {
+                                type: 'password',
+                                name: 'password',
+                                message: 'Password',
+                                validate: (answer) => !!answer
+                            }
+                        ]);
+                        service.defaults.auth = {
+                            username,
+                            password
+                        };
+                    } else {
+                        throw error;
+                    }
+                }
+            }
+    
+            this.answers.mainEntity = (await this.prompt({
+                type: "input",
+                name: "mainEntity",
+                message: "What entity should be used for the new page?",
+                validate: (s) => {
+                    if (/^\d*[a-zA-Z][a-zA-Z0-9]*$/g.test(s)) {
```

Comment:
same like above with ignore case

Reply:
I created a utils function and reused in the two places I need it as well as where I copied it from https://github.com/ui5-community/generator-ui5-project/pull/42/commits/adc10cdf027189d05c772d521167466b01ee63d1